sdk/core/src/main/java/com/klaviyo/core/utils/KlaviyoThreadHelper.kt

```diff
@@ -0,0 +1,26 @@
+package com.klaviyo.core.utils
+
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Looper
+
+/**
+ * Abstraction of our interactions with handlers/threads for isolation purposes
+ * @deprecated Use [com.klaviyo.core.Registry.threadHelper] instead
+ */
+object KlaviyoThreadHelper : ThreadHelper {
+    override fun getHandler(looper: Looper) = Handler(looper)
+    override fun getHandlerThread(name: String?) = HandlerThread(name)
+    override fun runOnUiThread(block: () -> Unit) {
+        val mainLooper = Looper.getMainLooper()
+
+        if (mainLooper == Looper.myLooper()) {
+            // Already on main thread, run immediately
+            block()
+        } else {
+            // Post to main thread
+            getHandler(mainLooper)
+                .post(block)
+        }
+    }
+}
```

Comment:
Simple threading helper, particularly convenient for mocking in unit tests.

This version of `runOnUiThread` is a little cleaner than earlier approaches I had. It also avoids re-dispatching if we are already on the main thread.
- Relying on`view.post` means the posted runner will only execute if the view is presented
- Reying on `currentActivity` requires mildly annoying optional chaining. 
- Both of these would re-dispatch even if called from the main thread.